encoding/string/encoding.go

```diff
@@ -0,0 +1,46 @@
+package string
+
+import "io"
+
+// Encoder encodes string values.
+type Encoder interface {
+	// Encode a collection of strings.
+	// Callers should explicitly call `Reset` before
+	// subsequent call to `Encode`.
+	Encode(Iterator) error
+	// Bytes returns the encoded bytes of the encoder.
+	// The format of the byte slice is as follows:
+	//   - version:                 1 byte
+	//   - metadata length bytes:   VInt
+	//   - metadata bytes:          Variable bytes
+	//   - data bytes:              Variable bytes
+	Bytes() []byte
```

Comment:
Can you remove the encoding format here? Probably implementation detail that shouldn't be part of the API docs.

Reply:
👍